From 472278fae5695495e2ad6289a97976bddc9a7b92 Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Wed, 21 Mar 2012 13:23:52 +0800
Subject: [PATCH 256/498] ENGR00177366 The logcat information was not correct
 and redundant.

Removed the redundant information and checked the memory status before
release it.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index c9c39c7..197eba2 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -723,7 +723,7 @@ namespace android {
         android_native_buffer_t *buf;
         private_handle_t *handle;
         for(unsigned int i = 0; i < mCaptureBufNum; i++) {
-            //if(mCaptureBuffers[i].buf_state == WINDOW_BUFS_DEQUEUED) {    
+            if(mCaptureBuffers[i].buf_state == WINDOW_BUFS_DEQUEUED) {
                 buf = (android_native_buffer_t *)mCaptureBuffers[i].native_buf;
                 if(mCaptureBuffers[i].virt_start != NULL) {
                     handle = (private_handle_t *)buf->handle;
@@ -732,7 +732,9 @@ namespace android {
                 if(buf != NULL) {
                     mNativeWindow->cancel_buffer(mNativeWindow, &buf->handle);
                 }
-            //}
+            }
+            else
+                continue;
             mCaptureBuffers[i].buf_state = WINDOW_BUFS_INVALID;//WINDOW_BUFS_QUEUED;
             mCaptureBuffers[i].refCount = 0;
             mCaptureBuffers[i].native_buf = NULL;
@@ -2218,11 +2220,10 @@ Pic_out:
     
     int CameraHal::encodeframeThreadWrapper()
     {
-        CAMERA_HAL_LOG_FUNC;
+        //CAMERA_HAL_LOG_FUNC;
         status_t ret = NO_ERROR;
         while(1) {
             if(mExitEncodeThread) {
-                CAMERA_HAL_ERR("%s: encode thread exit normally", __FUNCTION__);
                 return 0;
             }
             ret = encodeframeThread();
-- 
1.8.0

